### DISCUSSION TITLE
Data, Code and Visualization - valueBoxes and Sparklines

### DISCUSSION THREAD

**Discussion by giscus** [jbkunst/blog#2] [General]
# Data, Code and Visualization - valueBoxes and Sparklines

The value boxes are a good way to get to the point quickly in a dashboard showing the relevant value with a big font and eye-catching color. But sometimes it would be better show that relevant number with a related text, sparkline or information to give context of that value, or just to compare it with the previoues ones. In this post we’ll show how to add sparklines to valueBox in shinydashboard and other related packages.

http://jkunst.com/blog/posts/2020-06-26-valuebox-and-sparklines/

**Comment by sanjmeh** [comment]
this is great! Do you have a package that provides these functions?

**Reply by jbkunst** [reply]
Hi @sanjmeh ,

Now {bslib} package provide the `value_box` function with a `showcase` parameter which provides similar functionality.

Reference: https://rstudio.github.io/bslib/articles/value-boxes.html

**Comment by sactyr** [comment]
Hey, thank you for this. I am interested in capturing mouse events (specially clicks) if one clicks on the value box, is there a way to do this? I tried changing the class in the last div to `action-button`:
from
```
div(
    class = if (!is.null(width)) paste0("col-sm-", width), 
    boxContent
    )
```
to
```
div(
    class = "action-button",
    id = "testvb",
    boxContent
    )
```
and then using an `observeEvent` to capture the clicks:
```
observeEvent(input$testvb, {
    print(input$testvb)
 })
```
But this doesn't quite work; I am not able to print any clicks to console. Looking at the underlying html in browser's inspect, the id "testvb" is not visible at all, so I think I am not changing the class/id correctly. Any help is appreciated, cheers

**Reply by sactyr** [reply]
ok I literally solved it after seconds of posting the above. Turns out I needed to introduce `class` and `id` within boxContent i.e.
```
boxContent <- div(
    class = paste0("action-button small-box bg-", color), 
    id = id,
    ...
  )
```
Thank you once again :)

**Comment by jbkunst** [comment]
Nice idea, @sactyr! Glad the implementation is easy.
